test(Complete insuffisance test E2E)

lomamech opened the pull request:
test(Complete insuffisance test E2E
- Complete test E2E for the research with the inventorie group uuid
in Cash registry, Patient Registry and Invoice Registry
- Update some function the test E2E
- Update API for cash and patient invoice

closes #1733

lomamech committed:
test(Complete insuffisance test E2E)

- Complete test E2E for the research with the inventorie group uuid
in Cash registry, Patient Registry and Invoice Registry
- Update some function the test E2E
- Update API for cash and patient invoice

jniles commented on client/src/modules/cash/payments/templates/search.modal.html:
```diff
@@ -55,6 +55,31 @@
+            </span>
+
+            <!-- @todo make data driven selects components to handle loading and error states -->
+            <select
```
In most places, I think we use a `ui-select` for debtor groups.  We should probably use the same here.

jniles commented on client/src/modules/cash/payments/templates/search.modal.html:
```diff
@@ -55,6 +55,31 @@
+
+            <!-- @todo make this a component - it is used everywhere. Calls back method
+                 `clear` and resets the id passed to the component -->
+            <span style="display:inline-block;" class="pull-right">
```
This should use `bhClear`.  See #1702.

jniles commented on test/end-to-end/cash/registry.spec.js:
```diff
@@ -70,6 +71,12 @@ function CashPaymentsRegistryTests() {
     GU.expectRowCount('payment-registry', PAYMENT_INSIDE_REGISTRY);
   });
 
+  it('finds all payments for creditor group Second Test Debtor Group', () => {
```
:+1:

jniles commented on client/src/modules/invoices/registry/search.modal.html:
```diff
@@ -59,6 +59,32 @@
+
+        <!-- @todo make this a component - it is used everywhere. Calls back method
+             `clear` and resets the id passed to the component -->
+        <span style="display:inline-block;" class="pull-right">
```
Same for here!  You can also remove that `@todo` ... we implemented it in `bhClear`.

jniles reviewed (changes requested):
@lomamech, very nicely implemented tests.  This was a good catch.

We already have a component for clearing search form inputs - it's `bhClear` and you should use it here.

Most places use a `ui-select` for debtor groups (e.g. Patient Registration), so it would probably be best to use one here.

After that, LGTM!

lomamech committed:
perf(Test e2e and best practice)

- Use ui-select instead select
- Use componant bhClear and fix issue 1702
- Update test E2E

closes #1702

jniles commented on client/src/modules/inventory/list/modals/search.modal.html:
```diff
@@ -18,36 +18,26 @@
-            <i class="fa fa-eraser"></i> <span translate>FORM.BUTTONS.CLEAR</span>
-          </a>
-        </span>
+        <bh-clear on-clear="ModalCtrl.clear('group_uuid')"></bh-clear>
```
:+1: Much cleaner!

jniles reviewed (approved):
LGTM
